p5/visualTests/sanity_testing.py

```diff
@@ -0,0 +1,39 @@
+import subprocess
+import os
+from subprocess import run, PIPE
+
+dirname = os.path.dirname(os.path.realpath(__file__))
+
+test_names_2d = [f'{dirname}/sanityTests/2DSanityTests/{name}' for name in
+                 os.listdir(os.path.join(dirname, 'sanityTests/2DSanityTests'))
+                 if name not in ['__init__.py', '__pycache__']]
+
+test_names_3d = [f'{dirname}/sanityTests/3DSanityTests/{name}' for name in
+                 os.listdir(os.path.join(dirname, 'sanityTests/3DSanityTests'))
+                 if name not in ['__init__.py', '__pycache__']]
+
+print('============================= Running 2D tests =============================')
+for name in test_names_2d:
+    print('Running', name)
+    try:
+        # Run each test for 2s then raise a timeout
+        process = run(['python', name], stderr=PIPE, encoding='utf-8', timeout=2)
+    except subprocess.TimeoutExpired:
+        pass
+    else:
+        # If any error was found raise and exception
+        if process.stderr:
+            raise Exception(f'{process.stderr} in {name}')
+
+print('============================= Running 3D tests =============================')
+for name in test_names_3d:
+    print('Running', name)
+    try:
+        # Run each test for 2s then raise a timeout
+        process = run(['python', name], stderr=PIPE, encoding='utf-8', timeout=2)
+    except subprocess.TimeoutExpired:
+        pass
+    else:
```

Comment:
One last detail: I think we'll expose more exception details if we delete the else clause here.

Reply:
Oops I forgot that you are running it in a subprocess - yeah we should keep this else clause.